src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,232 @@
+	}
+
+	/**
+	 * Gets the length of the arraylist.
+	 *
+	 * @return The length of the arraylist
+	 */
+	public int size() {
+		return this.list.size();
+	}
+
+	/**
+	 * Clears the arraylist.
+	 */
+	public void clear() {
+		this.list.clear();
+	}
+
+	/**
+	 * marks task with an X
+	 *
+	 * @param task
+	 */
+	public static void markString(Task task) {
+		task.mark();
+		System.out.println("-----------------------------------------------");
+		System.out.println("Nice! I've marked this task as done:");
+		System.out.println(task);
+		System.out.println("-----------------------------------------------");
+	}
+
+	/**
+	 * Unmarks task by removing the X.
+	 *
+	 * @param task
+	 */
+	public static void unMarkString(Task task) {
+		task.unMark();
+		System.out.println("-----------------------------------------------");
+		System.out.println("OK, I've marked this task as not done yet:");
+		System.out.println(task);
+		System.out.println("-----------------------------------------------");
+	}
+
+	/**
+	 * To mark or unmark the task.
+	 *
+	 * @param update The action to be done.
+	 * @param parts The string response broken down into its keywords.
+	 * @throws DukeException If the keywords are missing or invalid.
+	 */
+	public void updateTask(Parser.Update update, String[] parts ) throws DukeException {
+		String part2;
+		switch (update) {
+		case MARK:
+			/** when there is no number declared */
+			if (parts.length <= 1) {
+				throw new DukeException("Please tell me what to mark!");
+			}
+			part2 = parts[1];
+			if (isInteger(part2)) {
+				int number = Integer.parseInt(part2);
+				/** To check if the integer is valid */
+				if (this.list.size() < number || number <= 0) {
+					throw new DukeException("There's no such task to mark!");
+				} else {
+					markString(this.list.get(number - 1));
+				}
+			} else {
+				throw new DukeException("I don't know which to mark!");
+			}
+			break;
+		case UNMARK:
+			/** when there is no number declared */
+			if (parts.length <= 1) {
+				throw new DukeException("Please tell me what to unmark!");
+			}
+			part2 = parts[1];
+			if (isInteger(part2)) {
+				int number = Integer.parseInt(part2);
+				/** To check if the integer is valid */
+				if (this.list.size() < number || number <= 0) {
+					throw new DukeException("There's no such task to unmark!");
+				} else {
+					unMarkString(this.list.get(number - 1));
+				}
+			} else {
+				throw new DukeException("I don't know which to unmark!");
+			}
+			break;
+		}
+	}
+
+	/**
+	 * Adds task into the arraylist.
+	 *
+	 * @param task The task added into the arraylist.
+	 */
+	public void addDetailedTask(Task task) {
+		System.out.println("-----------------------------------------------");
+		System.out.println("Got it. I've added this task:");
+		System.out.println(task);
+		System.out.println("Now you have " + this.list.size() +" tasks in the list.");
+		System.out.println("-----------------------------------------------");
```

Comment:
Perhaps you can consider migrating these print statements into Ui.java as well for a better organisation of classes.